net/http.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package net
+
+import (
+	"errors"
+	"net"
+	"net/http"
+	"strings"
+)
+
+// GetClientIPFromRequest retrieves the client's IP address from the request.
+// This checks common headers that forward the client IP, falling back to the
+// request's `RemoteAddr`.
+func GetClientIPFromRequest(r *http.Request) (net.IP, error) {
+	clientIP, err := func() (string, error) {
+		// `Forwarded` (RFC 7239).
+		forwardedHeader := r.Header.Get("Forwarded")
+		if forwardedHeader != "" {
+			parts := strings.Split(forwardedHeader, ",")
```

Comment:
This is acceptable for this PR, but you need to update it later to use proper libraries to parse this format.
I believe https://pkg.go.dev/net/textproto#Reader.ReadMIMEHeader will give you the list of values, which you can then split. We need to figure out what the format is based on the RFC. We can look into what the net lib does.